dashboard/lib/pd/payment/payment_calculator.rb

```diff
@@ -0,0 +1,46 @@
+module Pd::Payment
+  class PaymentCalculator
+    include Singleton
+    MINIMUM_PUZZLES_FOR_CSF_QUALIFICATION = 10
+    MINIMUM_QUALIFIED_TEACHERS_FOR_CSF = 10
+    PAYMENT_PER_TEACHER_CSF = 50
+
+    # Under these payment terms we can have a *much* simpler workshop payment calculator
+    def calculate(workshop)
+      raise "Workshop #{workshop.id} is not ended - cannot pay" if
+          workshop.state != Pd::Workshop::STATE_ENDED
+
+      if workshop.course == Pd::Workshop::COURSE_CSF
+        calculate_csf_payment(workshop)
+      else
+        calculate_non_csf_payment(workshop)
```

Comment:
This should be worded in the affirmative, something like `general_payment` or `default_payment`. CSF is the special case. This is the standard case. It also will make things less awkward if we ever have to add another special case.